server/src/processing/TableOfContentsDetectionModule/detection-methods.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2020 AXA Group Operations S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BoundingBox, Paragraph, Word } from './../../types/DocumentRepresentation';
+
+export const threshold = 0.4;
+
+export function TOCDetected(p: Paragraph): boolean {
+  return Object.values(detectionMethods).some(method => method(p));
+}
+
+const detectionMethods = {
+  /*
+    searches for text finishing in numbers in the right 10% width area of the BBox
+  */
+  endsWithNumber: (p: Paragraph): boolean => {
+    const w = p.width * 0.1;
+    const intersectionBox = new BoundingBox(p.right - w, p.top, w, p.height);
+    const wordsInsideIntersection =
+      p.getWords()
+        .filter(word => BoundingBox.getOverlap(word.box, intersectionBox).box1OverlapProportion > 0)
+        .filter(word => !isSeparator(word));
+
+    return wordsInsideIntersection.filter(isNumber).length > Math.floor(wordsInsideIntersection.length * 0.5);
+  },
+  hasPageNKeyword: (p: Paragraph): boolean => {
+    return new RegExp(/^pag.* (\d+) (.+)/gi).test(p.toString());
+  },
+};
+
+// TODO maybe handle this in a different way
+const tocKeywords = [
+  'contents',
+  'index',
+  'table of contents',
+  'contenidos',
+  'indice',
+  'índice',
+  'tabla de contenidos',
+];
```

Comment:
Do you think has sense to add those keywords to new module defaultConfig parameter ??

Reply:
done in https://github.com/axa-group/Parsr/pull/310/commits/477ceea64e51eb38cb49b1d3758a290fea3dcb28